packages/core/lib/__tests__/use-action-history.spec.tsx

```diff
@@ -0,0 +1,192 @@
+    const actions = [
+      { rewind: jest.fn(), forward: jest.fn() },
+      { rewind: jest.fn(), forward: jest.fn() },
+      { rewind: jest.fn(), forward: jest.fn() },
+    ];
+
+    beforeEach(() => {
+      actions.forEach(({ rewind, forward }) => {
+        act(() =>
+          renderedHook.result.current.record({
+            rewind,
+            forward,
+          })
+        );
+      });
+    });
+
+    test("rewind function of action should be execute", () => {
+      actions.reverse().forEach(({ rewind }) => {
+        act(() => renderedHook.result.current.rewind());
+        expect(rewind).toBeCalled();
+      });
+    });
+
+    test("canRewind should be true until before last rewind", () => {
+      actions.reverse().forEach((_, i) => {
+        act(() => renderedHook.result.current.rewind());
+        expect(renderedHook.result.current.canRewind).toBe(
+          i !== actions.length - 1
+        );
+      });
+    });
+  });
+
+  describe("when run forward all with action histories (after all rewinded)", () => {
+    const actions = [
+      { rewind: jest.fn(), forward: jest.fn() },
+      { rewind: jest.fn(), forward: jest.fn() },
+      { rewind: jest.fn(), forward: jest.fn() },
+    ];
+
+    beforeEach(() => {
+      actions.forEach(({ rewind, forward }) => {
+        act(() =>
+          renderedHook.result.current.record({
+            rewind,
+            forward,
+          })
+        );
+      });
+    });
+
+    beforeEach(() => {
+      actions.forEach(() => {
+        act(() => renderedHook.result.current.rewind());
+      });
+    });
+
+    test("forward function of action should be execute", () => {
+      actions.forEach(({ forward }) => {
+        act(() => renderedHook.result.current.forward());
+        expect(forward).toBeCalled();
+      });
+    });
+
+    test("canForward should be true until before last forward", () => {
+      actions.forEach((_, i) => {
+        act(() => renderedHook.result.current.forward());
+        expect(renderedHook.result.current.canForward).toBe(
+          i !== actions.length - 1
+        );
+      });
+    });
+  });
+
+  describe("in the middle of history", () => {
+    const actions = [
+      { rewind: jest.fn(), forward: jest.fn() },
+      { rewind: jest.fn(), forward: jest.fn() },
+      { rewind: jest.fn(), forward: jest.fn() },
+    ];
+
+    beforeEach(() => {
+      actions.forEach(({ rewind, forward }) => {
+        act(() =>
+          renderedHook.result.current.record({
+            rewind,
+            forward,
+          })
+        );
+      });
+      act(() => renderedHook.result.current.rewind());
+    });
+
+    test("recent histories are rewritten when recording a new action ", () => {
+      const newAction = { rewind: jest.fn(), forward: jest.fn() };
+      act(() => renderedHook.result.current.record(newAction));
+
+      expect(renderedHook.result.current.canForward).toBe(false);
+    });
+  });
+});
```

Comment:
Could you add this to the CI file so the tests run as part of this build?

```yml
      - name: Run tests
        run: yarn test
```

`.github/workflows/ci.yml` around L30 would be great